projects/graph/src/graph.py

```diff
@@ -1,9 +1,59 @@
 """
 Simple graph implementation compatible with BokehGraph class.
 """
+class Queue:
+    def __init__(self):
+        self.queue = []
+    def enqueue(self, value):
+        self.queue.append(value)
+    def dequeue(self):
+        if (self.size()) > 0:
+            return self.queue.pop(0)
+        else:
+            return None
+    def size(self):
+        return len(self.queue)
 
+class Stack:
+    def __init__(self):
+        self.stack = []
+    def push(self, value):
+        self.stack.append(value)
+    def pop(self):
+        if (self.size()) > 0:
+            return self.stack.pop()
+        else:
+            return None
+    def size(self):
+        return len(self.stack)
 
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+    def add_vertex(self,vertex):#Vertices are keys in dictionary
+        self.vertices[vertex] = set()
+    def add_edge(self,start,end):
+        if start in self.vertices and end in self.vertices:
+            self.vertices[start].add(end)
+            self.vertices[end].add(start)
+        else:
+            raise IndexError("That vertex does not exist!")
+    def bft(self, starting_node):
+        q = Queue()
+        q.enqueue(starting_node)
+        visited = []
+        while q.size() > 0:
+            visited_item = q.pop(0)
+            if visited_item not in visited:
+                visited.append(visited_item)
+                q.extend(list(visited_item.add_edge))
```

Comment:
This line is interesting, why the choice to use extend  and visited_item.add_edge? It seems that it would be better to use the enqueue function of your Queue class, and it's not clear to me that .add_edge() is appropriate here, maybe I'm misunderstanding?